packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -247,7 +262,9 @@ class FlutterDevice {
           getSkSLMethod: getSkSLMethod,
           printStructuredErrorLogMethod: printStructuredErrorLogMethod,
           device: device,
-        );
+        ), device.dds.done.whenComplete(() {
+          throw Exception('DDS shut down too early');
```

Comment:
Can you add some more context to this exception? this case does indicate an error in dds right?

Reply:
It simply indicates that DDS has shutdown, not necessarily whether or not it encountered an error. However, if we've shutdown DDS before the VM service connection is established it means something has gone wrong, likely the VM service disappearing. It's described in the comment block above the try block.